### DISCUSSION TITLE
How to get webhook body (unknown/dynamic schema) without ?raw=true

### DISCUSSION THREAD

**Discussion by tennox** [windmill-labs/windmill#4270] [Questions]
I don't see a way to get webhook payload without adding `?raw=true` to the URL on the calling side, I would need to know the schema of the payload beforehand, which I don't, or also in cases where the schema might have different variants (e.g. GitLab webhook push event vs. deployment event - different fields).

### Edit: related issue - Headers?
I also can't access headers without query arg `&include_header=X-Gitlab-Event` or setting an env var (on the instance?) - which I guess it's a security related thing? That jobs can't access e.g. Auth headers of the windmill instance? :thinking:

**Comment by rubenfiszel** [comment]
On latest, we added the query arg `wrap_body=true `

This is useful when the payload is not not known in advance and you want to handle it in your script. e.g:

Python:

```python
def main(body: Any):
	print(body)
```

Typescript:

```typescript
export async function main(body: any) {
	console.log(body);
}
```

**Reply by tennox** [reply]
Ok, that saves me the parsing step, thanks 🙏

But does it mean there's good reasons not to make this possible without query param args?
Or more complicated to get the UX right? 🤔
